apps/betterangels-backend/shelters/types.py

```diff
@@ -0,0 +1,211 @@
+    name: Optional[StorageChoices]
+
+
+@strawberry_django.type(Pet)
+class PetType:
+    name: Optional[PetChoices]
+
+
+@strawberry_django.type(Parking)
+class ParkingType:
+    name: Optional[ParkingChoices]
+
+
+@strawberry_django.type(ImmediateNeed)
+class ImmediateNeedType:
+    name: Optional[ImmediateNeedChoices]
+
+
+@strawberry_django.type(GeneralService)
+class GeneralServiceType:
+    name: Optional[GeneralServiceChoices]
+
+
+@strawberry_django.type(HealthService)
+class HealthServiceType:
+    name: Optional[HealthServiceChoices]
+
+
+@strawberry_django.type(TrainingService)
+class TrainingServiceType:
+    name: Optional[TrainingServiceChoices]
+
+
+@strawberry_django.type(EntryRequirement)
+class EntryRequirementType:
+    name: Optional[EntryRequirementChoices]
+
+
+@strawberry_django.type(City)
+class CityType:
+    name: Optional[CityChoices]
+
+
+@strawberry_django.type(SPA)
+class SPAType:
+    name: Optional[SPAChoices]
+
+
+@strawberry_django.type(ShelterProgram)
+class ShelterProgramType:
+    name: Optional[ShelterProgramChoices]
+
+
+@strawberry_django.type(Funder)
+class FunderType:
+    name: Optional[FunderChoices]
+
+
+@strawberry_django.type(Shelter)
+class ShelterType:
+    id: ID
+    accessibility: List[AccessibilityType]
+    additional_contacts: List[ContactInfoType]
+    bed_fees: Optional[str]
+    cities: List[CityType]
+    city_council_district: auto
+    curfew: auto
+    demographics: List[DemographicType]
+    demographics_other: auto
+    description: str
+    email: auto
+    entry_info: Optional[str]
+    entry_requirements: List[EntryRequirementType]
+    exterior_photos: List[ShelterPhotoType]
+    funders: List[FunderType]
+    funders_other: auto
+    general_services: List[GeneralServiceType]
+    health_services: List[HealthServiceType]
+    immediate_needs: List[ImmediateNeedType]
+    interior_photos: List[ShelterPhotoType]
+    location: ShelterLocationType
+    max_stay: auto
+    name: auto
+    on_site_security: auto
+    organization: Optional[OrganizationType]
+    other_rules: Optional[str]
+    other_services: Optional[str]
+    overall_rating: auto
+    parking: List[ParkingType]
+    pets: List[PetType]
+    phone: PhoneNumberScalar  # type: ignore
+    program_fees: Optional[str]
+    room_styles: List[RoomStyleType]
+    room_styles_other: auto
+    shelter_programs: List[ShelterProgramType]
+    shelter_programs_other: auto
+    shelter_types: List[ShelterKindType]
+    shelter_types_other: auto
+    spa: List[SPAType]
+    special_situation_restrictions: List[SpecialSituationRestrictionType]
+    status: auto
+    storage: List[StorageType]
+    subjective_review: Optional[str]
+    supervisorial_district: auto
+    total_beds: auto
+    training_services: List[TrainingServiceType]
+    website: auto
+
+    @strawberry_django.field
+    def hero_image(self, root: Shelter) -> Optional[str]:
+        if exterior_photos := getattr(root, "prefetched_exterior_photos", None):
```

Comment:
I think this can be simplified to

```python
    photos = chain(
        getattr(root, "prefetched_exterior_photos", []),
        getattr(root, "prefetched_interior_photos", [])
    )
    
    photo = next(photos, None)
    return str(photo.file.url) if photo else None
 ```

Reply:
Probably also want to put a comment here to say this is meant to be temporary until the hero_image is added back

Reply:
gpt sure loves its itertools chain. recently tried telling me that chaining two db hits would reduce query count 🙄 . but ya, this is better